### DISCUSSION TITLE
How to write a simple program in easy way?

### DISCUSSION THREAD

**Discussion by paulocoutinhox** [ggml-org/llama.cpp#3563] [Q&A]
Hi,

How can i write a `simple` C++ program in the most easy way that use `llama.cpp` to use a local model and get their prompt in sync way? 

Example of pseudo-code:

```
initialize_llama_cpp()

string response = get_prompt_for("./my-model.bin", "what is the biggest planet?") // here it will wait until full response finished be produced

print(response)
```

Thanks for any help.

**Comment by staviq** [comment]
[examples/simple/simple.cpp](https://github.com/ggerganov/llama.cpp/blob/master/examples/simple/simple.cpp)

**Reply by paulocoutinhox** [reply]
Thanks!
